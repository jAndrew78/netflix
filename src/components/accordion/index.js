import React, { useState } from 'react';


export default function Accordion({ children, ...restProps }) {
    <Container {...restProps}>
        <Inner>{children}</Inner>
    </Container>
}

Accordion.Title = function AccordionTitle({ children, ...restProps }) {
    return <Title {...restProps}>{ children }</Title>;
}

Accordion.Frame = function AccordionFrame({ children, ...restProps }) {
    return <Frame {...restProps}>{ children }</Frame>;
}

Accordion.Item = function AccordionItem({ children, ...restProps }) {
    return <Item {...restProps}>{ children }</Item>;
}

Accordion.Body = function AccordionBody({ children, ...restProps }) {
    return <Body {...restProps}>{ children }</Body>;
}

