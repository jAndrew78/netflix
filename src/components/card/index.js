import React from 'react';
import { Container, Group, Title, SubTitle, Text, } from './styles/card';


export default function Card({ children, ...restProps }) {
    return <Container {...restProps}>{children}</Container>;
}

Card.Group = function CardGroup({ children, ...restProps }) {
    return <Group {...restProps}>{children}</Group>
}

Card.Title = function CardTitle({ children, ...restProps }) {
    return <Title {...restProps}>{children}</Title>
}

Card.SubTitle = function CardSubTitle({ children, ...restProps }) {
    return <SubTitle {...restProps}>{children}</SubTitle>
}

Card.Text = function CardText({ children, ...restProps }) {
    return <Text {...restProps}>{children}</Text>
}
