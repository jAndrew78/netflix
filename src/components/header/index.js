import React, { useState } from 'react';
import { Link as ReactRouterLink } from 'react-router-dom';

import { 
    Background, 
    Container, 
    Group, 
    Feature, 
    Logo, 
    ButtonLink, 
    Picture, 
    Profile,
    Dropdown,
    DropdownBreak,
    Break, 
    Text, 
    FeatureCallOut, 
    Link, 
    Search,
    SearchIcon,
    CloseIcon,
    SearchInput,
} from './styles/header';


export default function Header({ bg = true, children, ...restProps }) {
    return bg ? <Background {...restProps}>{children}</Background> : children;
}

Header.Frame = function HeaderFrame({ children, ...restProps }) {
    return <Container {...restProps}>{children}</Container>
}

Header.Group = function HeaderGroup({ children, ...restProps }) {
    return <Group {...restProps}>{children}</Group>
}

Header.Feature = function HeaderFeature({ children, ...restProps }) {
    return <Feature {...restProps}>{children}</Feature>
}

Header.Logo = function HeaderLogo({ to, ...restProps }) {
    return (
        <ReactRouterLink to={to}>
            <Logo {...restProps} />
        </ReactRouterLink>
    )
}

Header.ButtonLink = function HeaderButtonLink({ children, ...restProps }) {
    return <ButtonLink {...restProps}>{children}</ButtonLink>
}

Header.Picture = function HeaderPicture({ src, ...restProps }) {
    return <Picture {...restProps} src={`/images/users/${src}.png`} />
}

Header.Profile = function HeaderProfile({ children, ...restProps }) {
    return <Profile {...restProps}>{children}</Profile>
}

Header.Search = function HeaderSearch({ searchTerm, setSearchTerm, ...restProps }) {
    const [searchActive, setSearchActive] = useState(false);

    return (
        <Search {...restProps}>
            <SearchIcon onClick={() => setSearchActive(searchActive => !searchActive)} active={searchActive}>
                <img src="/images/icons/search.png" alt="Search" />
            </SearchIcon>
            <SearchInput
                value={searchTerm}
                onChange={({ target }) => setSearchTerm(target.value)}
                placeholder="Search films and series"
                active={searchActive}
            />
            <CloseIcon onClick={() => setSearchActive(searchActive => !searchActive)} active={searchActive}>
                <img src="/images/icons/close-slim.png" alt="Close" />
            </CloseIcon>
        </Search>
    ) 
}

Header.Dropdown = function HeaderDropdown({ children, ...restProps }) {
    return <Dropdown {...restProps}>{children}</Dropdown>
}

Header.DropdownBreak = function HeaderDropdownBreak({ children, ...restProps }) {
    return <DropdownBreak {...restProps}>{children}</DropdownBreak>
}

Header.Break = function HeaderBreak({ children, ...restProps }) {
    return <Break {...restProps}>{children}</Break>
}

Header.Text = function HeaderText({ children, ...restProps }) {
    return <Text {...restProps}>{children}</Text>
}

Header.FeatureCallOut = function HeaderFeatureCallOut({ children, ...restProps }) {
    return <FeatureCallOut {...restProps}>{children}</FeatureCallOut>
}

Header.TextLink = function HeaderTextLink({ children, ...restProps }) {
    return <Link {...restProps}>{children}</Link>
}

