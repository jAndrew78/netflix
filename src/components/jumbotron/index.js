import React from 'react';
import { Inner, Container } from './styles/jumbotron';


export default function Jumbotron({ children, direction = 'row', ...restProps }) {    // get children, pass direction, then allowing any other props to be passed after
    return(
        <Inner direction={direction}>
            <p>Hello Jumbo</p>
        </Inner>
    );
}

Jumbotron.Container = function JumbotronContainer({ children, ...restProps }) {
    return <Container { ...restProps }>{children}</Container>;
}
