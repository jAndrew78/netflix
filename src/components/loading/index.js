import React from 'react';
import { Container } from './styles/loading';


export default function Loading({ children, ...restProps }) {
    return <Container {...restProps}>{children}</Container>
}