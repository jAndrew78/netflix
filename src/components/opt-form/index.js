import React from 'react';
import { Container, Input, Button, Text } from './styles/opt-form';


export default function OptForm({ children, ...restProps }) {    // get children, then allow any other props to be passed after
    return <Container {...restProps}>{ children }</Container>;
}

OptForm.Input = function OptFormInput({ ...restProps }) {
    return <Input {...restProps} />;
}

OptForm.Button = function OptFormButton({ children, ...restProps }) {
    return <Button {...restProps}>{ children }</Button>;
}

OptForm.Text = function OptFormText({ children, ...restProps }) {
    return <Text {...restProps}>{ children }</Text>;
}

