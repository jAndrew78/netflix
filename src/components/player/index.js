import React, { useState, useContext, createContext } from 'react';
import ReactDOM from 'react-dom';

import { Container, Button, Overlay, Inner, Close } from './styles/player';


// have to use context with styled components to be able to pass state around
export const PlayerContext = createContext();

export default function Player ({ children, ...restProps }) {
    const [showPlayer, setShowPlayer] = useState(false);

    return (
        <PlayerContext.Provider value={{ showPlayer, setShowPlayer }}>
            <Container {...restProps}>{children}</Container>
        </PlayerContext.Provider>
    ) 
}

Player.Video = function PlayerVideo({ src, ...restProps }) {
    const { showPlayer, setShowPlayer} = useContext(PlayerContext);

    return showPlayer ? ReactDOM.createPortal(
        <Overlay {...restProps}> 
            <Inner>
                <video id="netflix-player" controls>
                    <source src={src} type="video/mp4" />
                </video>
                <Close onClick={() => setShowPlayer(false)}> 
                    <img src="/images/icons/close.png" alt="Close" />
                </Close>
            </Inner>
        </Overlay>,
        document.body
    ) : null;
}

Player.Button = function PlayerButton({ ...restProps }) {
    const { showPlayer, setShowPlayer } = useContext(PlayerContext);

    return (
        <Button 
            onClick={() => setShowPlayer(showPlayer => !showPlayer)}>
            {/* use below to remove showPlayer error above, but may cause an error in rare cases */}
            {/* onClick={() => setShowPlayer(!showPlayer)}> */}
            <span>&#9654; </span>
            Play
        </Button>
    );
}
