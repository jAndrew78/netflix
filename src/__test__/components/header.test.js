import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import { Header } from '../../components';
import { MemoryRouter } from 'react-router-dom';

// jest.mock('react-router-dom');

describe('<Header />', () => {
    it('renders the basic <Header /> with a background', () => {
        const { container, getByText, getByTestId } = render(
            <Header>
                <Header.Frame>
                    <Header.Logo src="/logo.svg" alt="Netflix" to="/" />
                    <Header.TextLink active="true" to="/" >Hello I am a link!</Header.TextLink>
                </Header.Frame>
            </Header>,
            { wrapper: MemoryRouter }
        );

        expect(getByText('Hello I am a link!')).toBeTruthy();
        expect(getByTestId('header-bg')).toBeTruthy();
        expect(container.firstChild).toMatchSnapshot();
    });

    it('renders the basic <Header /> without a background', () => {
        const { container, getByText, queryByTestId } = render(
            <Header bg={false}>
                <Header.Frame>
                    <Header.Logo src="/logo.svg" alt="Netflix" to="/" />
                    <Header.ButtonLink to="/" >Sign In</Header.ButtonLink>
                    <Header.TextLink active="false" to="/" >Hello I am a link!</Header.TextLink>
                </Header.Frame>
            </Header>,
            { wrapper: MemoryRouter }
        );

        expect(getByText('Hello I am a link!')).toBeTruthy();
        expect(queryByTestId('header-bg')).toBeFalsy();
        expect(container.firstChild).toMatchSnapshot();
    });

    it('renders the full <Header /> with a background', () => {
        const { container, getByText, getByTestId } = render(
            <Header src="joker2" dontShowOnSmallViewPort>
                <Header.Frame>
                    <Header.Group>
                        <Header.Logo src="/images/logo.svg" alt="Netflix" to="/" />

                        <Header.TextLink active={false} onClick={() => {}} to="/">
                            TV Shows
                        </Header.TextLink>

                        <Header.TextLink active onClick={() => {}} to="/">
                            Movies
                        </Header.TextLink>
                    </Header.Group>

                    <Header.Group>

                        <Header.Search
                            searchTerm="Joker"
                            setSearchTerm={() => {}}
                        />

                        <Header.Profile>
                            <Header.Picture src='/images/andrew.png' />

                            <Header.DropdownBreak />

                            <Header.Dropdown>
                                <Header.Group>
                                    <Header.Picture src='/images/andrew.png' />
                                    <Header.TextLink to="/">Andrew</Header.TextLink>
                                </Header.Group>

                                <Header.Group>
                                    <Header.TextLink to="/">
                                        Manage Profiles
                                    </Header.TextLink>
                                </Header.Group>

                                <Header.Break />

                                <Header.Group>
                                    <Header.TextLink to="/">
                                        Account
                                    </Header.TextLink>
                                </Header.Group>

                                <Header.Group>
                                    <Header.TextLink to="/">
                                        Help Center
                                    </Header.TextLink>
                                </Header.Group>
                                
                                <Header.Group>
                                    <Header.TextLink onClick={() => {}} to="/">
                                        Sign Out of Netflix
                                    </Header.TextLink>
                                </Header.Group>
                            </Header.Dropdown>

                        </Header.Profile>

                    </Header.Group>
                </Header.Frame>

                <Header.Feature>
                    <Header.FeatureCallOut>Watch Joker Now</Header.FeatureCallOut>
                    <Header.Text>
                        Troubled comedian Aurthur Fleck...
                    </Header.Text>
                    <Header.PlayButton><span>&#9654; </span>Play</Header.PlayButton>
                </Header.Feature>
            </Header>,
            { wrapper: MemoryRouter }
        );

        expect(getByTestId('search-input')).toBeTruthy();
        expect(getByTestId('search-input').value).toBe('Joker');
        fireEvent.change(getByTestId('search-input'), {
            target: { value: 'Simpsons' },
        })
        fireEvent.click(getByTestId('search-click'));
        fireEvent.click(getByTestId('search-close'));

        expect(getByText('TV Shows')).toBeTruthy();
        expect(getByText('Movies')).toBeTruthy();
        expect(getByText('Andrew')).toBeTruthy();
        expect(getByText('Manage Profiles')).toBeTruthy();
        expect(getByText('Account')).toBeTruthy();
        expect(getByText('Help Center')).toBeTruthy();
        expect(getByText('Watch Joker Now')).toBeTruthy();
        expect(getByText('Sign Out of Netflix')).toBeTruthy();
        expect(getByText('Troubled comedian Aurthur Fleck...')).toBeTruthy();
        expect(getByText('Play')).toBeTruthy();
        expect(container.firstChild).toMatchSnapshot();
    });
});