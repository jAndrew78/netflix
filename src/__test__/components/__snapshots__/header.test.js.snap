// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders the basic <Header /> with a background 1`] = `
<div
  class="header__Background-djltcn-4 bHQNpi"
  data-testid="header-bg"
>
  <div
    class="header__Container-djltcn-0 bHyAQr"
  >
    <a
      href="/"
    >
      <img
        alt="Netflix"
        class="header__Logo-djltcn-5 iXegay"
        src="/logo.svg"
      />
    </a>
    <p
      class="header__Link-djltcn-8 fydlpe"
      to="/"
    >
      Hello I am a link!
    </p>
  </div>
</div>
`;

exports[`<Header /> renders the basic <Header /> without a background 1`] = `
<div
  class="header__Container-djltcn-0 bHyAQr"
>
  <a
    href="/"
  >
    <img
      alt="Netflix"
      class="header__Logo-djltcn-5 iXegay"
      src="/logo.svg"
    />
  </a>
  <a
    class="header__ButtonLink-djltcn-6 cQKhOl"
    href="/"
  >
    Sign In
  </a>
  <p
    class="header__Link-djltcn-8 kHLvHs"
    to="/"
  >
    Hello I am a link!
  </p>
</div>
`;

exports[`<Header /> renders the full <Header /> with a background 1`] = `
<div
  class="header__Background-djltcn-4 ixUnOD"
  data-testid="header-bg"
  src="joker2"
>
  <div
    class="header__Container-djltcn-0 bHyAQr"
  >
    <div
      class="header__Group-djltcn-1 glnGbv"
    >
      <a
        href="/"
      >
        <img
          alt="Netflix"
          class="header__Logo-djltcn-5 iXegay"
          src="/images/logo.svg"
        />
      </a>
      <p
        class="header__Link-djltcn-8 czjQWe"
        to="/"
      >
        TV Shows
      </p>
      <p
        class="header__Link-djltcn-8 czjQWe"
        to="/"
      >
        Movies
      </p>
    </div>
    <div
      class="header__Group-djltcn-1 glnGbv"
    >
      <div
        class="header__Search-djltcn-14 dtnxmj"
      >
        <button
          class="header__SearchIcon-djltcn-15 brdaVb"
          data-testid="search-click"
        >
          <img
            alt="Search"
            src="/images/icons/search.png"
          />
        </button>
        <input
          class="header__SearchInput-djltcn-17 kmVBpR"
          data-testid="search-input"
          placeholder="Search films and series"
          value="Joker"
        />
        <button
          class="header__CloseIcon-djltcn-16 gAhmDQ"
          data-testid="search-close"
        >
          <img
            alt="Close"
            src="/images/icons/close-slim.png"
          />
        </button>
      </div>
      <div
        class="header__Profile-djltcn-12 WiGLP"
      >
        <button
          class="header__Picture-djltcn-7 epwiFt"
          src="/images/users//images/andrew.png.png"
        />
        <div
          class="header__DropdownBreak-djltcn-10 hjYeSy"
        />
        <div
          class="header__Dropdown-djltcn-9 keHjTV"
        >
          <div
            class="header__Group-djltcn-1 glnGbv"
          >
            <button
              class="header__Picture-djltcn-7 epwiFt"
              src="/images/users//images/andrew.png.png"
            />
            <p
              class="header__Link-djltcn-8 czjQWe"
              to="/"
            >
              Andrew
            </p>
          </div>
          <div
            class="header__Group-djltcn-1 glnGbv"
          >
            <p
              class="header__Link-djltcn-8 czjQWe"
              to="/"
            >
              Manage Profiles
            </p>
          </div>
          <div
            class="header__Break-djltcn-11 dcYWEf"
          />
          <div
            class="header__Group-djltcn-1 glnGbv"
          >
            <p
              class="header__Link-djltcn-8 czjQWe"
              to="/"
            >
              Account
            </p>
          </div>
          <div
            class="header__Group-djltcn-1 glnGbv"
          >
            <p
              class="header__Link-djltcn-8 czjQWe"
              to="/"
            >
              Help Center
            </p>
          </div>
          <div
            class="header__Group-djltcn-1 glnGbv"
          >
            <p
              class="header__Link-djltcn-8 czjQWe"
              to="/"
            >
              Sign Out of Netflix
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="header__Container-djltcn-0 header__Feature-djltcn-2 bHyAQr kdETnK"
  >
    <h2
      class="header__FeatureCallOut-djltcn-3 jLwNcK"
    >
      Watch Joker Now
    </h2>
    <p
      class="header__Text-djltcn-13 kHftXZ"
    >
      Troubled comedian Aurthur Fleck...
    </p>
    <button
      class="header__PlayButton-djltcn-18 dqwxtr"
    >
      <span>
        â–¶ 
      </span>
      Play
    </button>
  </div>
</div>
`;
